pkg/traceql/engine_metrics.go

```diff
@@ -557,6 +470,47 @@ func (s *StepAggregator) Length() int {
 	return 0
 }
 
+// InstantAggregator is similar to StepAggregator but always has only one interval.
+// It's used for instant queries where we only need a single data point.
+type InstantAggregator struct {
+	start, end uint64
+	vector     VectorAggregator
+}
+
+var _ RangeAggregator = (*InstantAggregator)(nil)
+
+func NewInstantAggregator(start, end uint64, innerAgg func() VectorAggregator) *InstantAggregator {
+	return &InstantAggregator{
+		start:  start,
+		end:    end,
+		vector: innerAgg(),
+	}
+}
+
+func (i *InstantAggregator) Observe(span Span) {
+	// if outside the range, skip
+	if ts := span.StartTimeUnixNanos(); ts < i.start || ts > i.end {
+		return
+	}
+	i.vector.Observe(span)
+}
+
+func (i *InstantAggregator) ObserveExemplar(float64, uint64, Labels) {
+	// instant query does not have exemplars, so we skip
```

Comment:
any particular reason we don't? i don't think you should add it here or anything but i would have guessed we return exemplars with instant queries if you had asked. doesn't seem like there's any technical blockers to doing it.

Reply:
Nah, just wondering. Definitely not something to change in this PR.